File: src/Database/Model/DTOs/KonfiguratorDtos.cs
using System;
using System.Collections.Generic;

namespace AasDemoapp.Database.Model.DTOs
{
    public class CreateConfiguredProductDto
    {
        public string Name { get; set; } = string.Empty;
        public double Price { get; set; }
        public List<CreateProductPartDto> Bestandteile { get; set; } = [];
    }

    public class CreateProductPartDto
    {
        public long KatalogEintragId { get; set; }
        public string Name { get; set; } = string.Empty;
        public double Price { get; set; }
        public int Amount { get; set; }
        public DateTime UsageDate { get; set; }
    }

    public class ConfigurationResponseDto
    {
        public bool Success { get; set; }
        public string Message { get; set; } = string.Empty;
        public ConfiguredProductDto? ConfiguredProduct { get; set; }
        public string? Error { get; set; }
    }
}

File: src/AasDemoapp/ClientApp/src/app/model/konfigurator-dtos.ts
// Konfigurator Request DTOs
export interface CreateConfiguredProduct {
  name: string;
  price: number;
  aasId?: string;
  globalAssetId?: string;
  bestandteile: CreateProductPart[];
}

export interface CreateProductPart {
  katalogEintragId: number;
  name: string;
  price: number;
  amount: number;
  usageDate: Date;
}

// Konfigurator Response DTOs
export interface ConfigurationResponse {
  success: boolean;
  message: string;
  configuredProduct?: import('./configured-product').ConfiguredProduct;
  error?: string;
}
